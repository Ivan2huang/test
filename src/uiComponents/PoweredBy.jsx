import React from 'react';
import { injectIntl } from 'react-intl';
import * as PropTypes from 'prop-types';

import { Box } from '@material-ui/core';

import Typography from './Typography';

import IMAGES from '../constants/images';

const PoweredBy = ({ intl }) => {
  return (
    <Box
      display="flex"
      alignItems="center"
      justifyContent={{
        xs: 'center',
        md: 'flex-end',
      }}
    >
      <Box mr={2}>
        <Typography type="style-8">
          {intl.formatMessage({
            id: 'footer.logoText',
            defaultMessage: 'Powered by',
          })}
        </Typography>
      </Box>
      <img src={IMAGES.POWERED_LOGO} alt="logo" />
    </Box>
  );
};

PoweredBy.propTypes = {
  intl: PropTypes.shape({}).isRequired,
};

export default injectIntl(PoweredBy);
