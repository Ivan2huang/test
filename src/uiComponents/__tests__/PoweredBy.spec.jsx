import React from 'react';
import { render } from '@testing-library/react';

import PoweredBy from '../PoweredBy';
import withTheme from '../../themes/withThemeProvider';
import withIntl from '../../i18n/withIntlProvider';

jest.mock('../Typography', () => props => (
  <div {...props}>Typography Component</div>
));

jest.mock('../../constants/config', () => ({
  defaultLanguage: 'en-HK',
}));

describe('PoweredBy UI Component', () => {
  it('should match snapshot', () => {
    const Component = withIntl(withTheme(PoweredBy));
    const { container } = render(<Component />);

    expect(container).toMatchSnapshot();
  });
});
