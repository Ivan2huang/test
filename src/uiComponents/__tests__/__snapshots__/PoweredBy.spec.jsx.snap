// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoweredBy UI Component should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <div
        type="style-8"
      >
        Typography Component
      </div>
    </div>
    <img
      alt="logo"
      src="/static/powered-logo.svg"
    />
  </div>
</div>
`;
